sdk/helper/ldaputil/client.go

```diff
@@ -553,7 +553,7 @@ func (c *Client) GetLdapGroups(cfg *ConfigEntry, conn Connection, userDN string,
 	if cfg.UseTokenGroups {
 		entries, err = c.performLdapTokenGroupsSearch(cfg, conn, userDN)
 	} else {
-		if paging, ok := conn.(PagingConnection); ok && cfg.MaximumPageSize >= 0 {
+		if paging, ok := conn.(PagingConnection); ok && *cfg.MaximumPageSize >= 0 {
```

Comment:
The reports in https://github.com/hashicorp/vault/issues/20416 make me wonder if enabling paging by default is the right choice. Many are encountering issues and disabling it by setting the value to -1. Prior to `MaximumPageSize` existing, paging was not used. IIUC, paging is now enabled by default for users who are upgrading. I wonder if that behavior change will continue to catch users off guard.

Alternatively, the default could be -1 which would preserve expected behavior for users who are upgrading. The paging behavior could be opt-in. Let me know if I'm not understanding this correctly 🙂

Reply:
That makes sense to me.  I am okay with that solution.

Reply:
Hang on a minute, is that really true?

Didn't this entire saga start, because a user had a case where their server refused the use of paging, and wanted it made optional?

Having paging on by default is absolutely a good user-friendly common-case behaviour, because if you turn off paging, you spontaneously run into problems when the size of the LDAP queries that Vault needs to perform, exceeds the LDAP server's size limits.

Reply:
Oh, and as far as I can see, everyone's picking -1 because it's an easy obvious constant - unlike the numeric value corresponding to math.MaxInt32. The people selecting -1 are unwittingly configuring a less-than-ideal setting.